### PR TITLE
Let NotificationOverlay move the background

### PR DESCRIPTION
**Pull request by Aergwyn**
The same behaviour as SettingsOverlay.

### CODE REVIEW

File: osu.Game/OsuGame.cs [hunk 1]
```diff
@@ -276,12 +278,27 @@ protected override void LoadComplete()
...
+                {
+                    case Visibility.Hidden:
+                        backgroundOffset += ToolbarButton.WIDTH / 2;
                         break;
                     case Visibility.Visible:
-                        intro.MoveToX(SettingsOverlay.SIDEBAR_WIDTH / 2, SettingsOverlay.TRANSITION_LENGTH, Easing.OutQuint);
+                        backgroundOffset -= ToolbarButton.WIDTH / 2;
                         break;
                 }
+                intro.MoveToX(backgroundOffset, NotificationOverlay.TRANSITION_LENGTH, Easing.OutQuint);
```
Comment (peppy) [inline osu.Game/OsuGame.cs]: this is really bad code duplication. just make an inline function and bind the same thing to both `StateChanged` events.

```
void overlayStateChanged()
{
    float offset = 0;
    if (notificationOverlay.State == Visible) offset -= width /2;
    if (settingsOverlay == Visible) offset += width /2;
}
```

**Review by peppy — APPROVED** [review]

**Review by peppy — CHANGES_REQUESTED** [review]
poops

**Comment by Aergwyn** [comment]
I can't dismiss stale reviews, you may look over it again.

File: osu.Game/OsuGame.cs [hunk 2]
```diff
@@ -272,19 +272,21 @@ protected override void LoadComplete()
...
-                switch (settings.State)
-                {
-                    case Visibility.Hidden:
-                        intro.MoveToX(0, SettingsOverlay.TRANSITION_LENGTH, Easing.OutQuint);
-                        break;
-                    case Visibility.Visible:
-                        intro.MoveToX(SettingsOverlay.SIDEBAR_WIDTH / 2, SettingsOverlay.TRANSITION_LENGTH, Easing.OutQuint);
-                        break;
-                }
+                float offset = intro.X;
```
Comment (peppy) [inline osu.Game/OsuGame.cs]: why not just start with 0?

File: osu.Game/OsuGame.cs [hunk 3]
```diff
@@ -272,19 +272,21 @@ protected override void LoadComplete()
...
+
+                if (settings.State == Visibility.Hidden || notifications.State == Visibility.Visible)
```
Comment (peppy) [inline osu.Game/OsuGame.cs]: I don't understand these conditionals

**Review by peppy — APPROVED** [review]

**Pull request merged**